lib-extra/src/main/java/com/diffplug/spotless/extra/config/UserArgumentException.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2016 DiffPlug
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.diffplug.spotless.extra.config;
+
+import javax.annotation.Nullable;
+
+/**
+ * Exceptions caused due to invalid user arguments.
+ */
+class UserArgumentException extends RuntimeException {
```

Comment:
Can we just use `IllegalArgumentException` and `Preconditions` class?  I don't see why they can't fulfill the needs of this class.

Reply:
This shall just ensure that the error message contains a value which can be recognized by the user, so that the user knows where something needs to be fixed. Furthermore it allows distinction between user config errors and internal errors. 
Thought it would be a neat idea, but I can remove it if you like. Let me know your decision.

Reply:
I prefer to be ruthless in minimizing the amount of code we have to maintain, and the amount of code others have to read to contribute.  I'd prefer to remove it.